caas/kubernetes/provider/k8s.go

```diff
@@ -1740,6 +1804,82 @@ func (k *kubernetesClient) WatchUnits(appName string) (watcher.NotifyWatcher, er
 	return k.newWatcher(w, appName, k.clock)
 }
 
+func (k *kubernetesClient) WatchUnitStart(appName string) (watcher.StringsWatcher, error) {
+	pods := k.client().CoreV1().Pods(k.namespace)
+	selector := applicationSelector(appName)
+	logger.Debugf("selecting units %q to watch", selector)
+	w, err := pods.Watch(v1.ListOptions{
+		LabelSelector:        selector,
+		Watch:                true,
+		IncludeUninitialized: true,
+	})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	podsList, err := pods.List(v1.ListOptions{
+		LabelSelector:        applicationSelector(appName),
+		IncludeUninitialized: true,
+	})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	providerID := func(pod *core.Pod) string {
```

Comment:
This functionality to get the provider id is used here but also in Units(). Can we have a helper function used in both places or else the logic is in danger of diverging.